src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidAccountManagerBroker.cs

```diff
@@ -0,0 +1,455 @@
+            using (_logger.LogMethodDuration())
+            {
+                // onActivityResult will receive the response for this activity.
+                // Launching this activity will switch to the broker app.
+
+                _logger.Verbose("Starting Android Broker interactive authentication. ");
+                Intent brokerIntent = await GetIntentForInteractiveBrokerRequestAsync(brokerRequest, _parentActivity).ConfigureAwait(false);
+
+                if (brokerIntent != null)
+                {
+                    _brokerHelper.LaunchInteractiveActivity(_parentActivity, brokerIntent);
+                }
+            }
+        }
+
+        private async Task<Intent> GetIntentForInteractiveBrokerRequestAsync(BrokerRequest brokerRequest, Activity callerActivity)
+        {
+            try
+            {
+                // Callback is not passed since it is making a blocking call to get
+                // intent. Activity needs to be launched from calling app
+                // to get the calling app's meta data if needed at BrokerActivity.
+
+                Bundle addAccountOptions = new Bundle();
+                addAccountOptions.PutString(BrokerConstants.BrokerAccountManagerOperationKey, BrokerConstants.GetIntentForInteractiveRequest);
+
+                Bundle accountManagerResult = await PerformAccountManagerOperationAsync(addAccountOptions).ConfigureAwait(false);
+
+                return CreateIntentFromInteractiveBundle(accountManagerResult, brokerRequest);
+            }
+            catch
+            {
+                _logger.Error("Error when trying to acquire intent for broker authentication. ");
+                throw;
+            }
+        }
+
+        private Intent CreateIntentFromInteractiveBundle(Bundle bundle, BrokerRequest brokerRequest)
+        {
+            if (bundle == null)
+            {
+                _logger.Info("Android account manager didn't return any results for interactive broker request. ");
+            }
+
+            Intent interactiventent = (Intent)bundle?.GetParcelable(AccountManager.KeyIntent);
+
+            //Validate that the intent was created successfully.
+            if (interactiventent != null)
+            {
+                _logger.Info("Intent created from BundleResult is not null. Starting interactive broker request. ");
+                // Need caller info UID for broker communication
+                interactiventent.PutExtra(BrokerConstants.CallerInfoUID, Binder.CallingUid);
+            }
+            else
+            {
+                _logger.Info("Intent created from BundleResult is null. ");
+                throw new MsalClientException(MsalError.NullIntentReturnedFromAndroidBroker, MsalErrorMessage.NullIntentReturnedFromBroker);
+            }
+
+            return CreateInteractiveBrokerIntent(brokerRequest, interactiventent);
+        }
+
+        private Intent CreateInteractiveBrokerIntent(BrokerRequest brokerRequest, Intent brokerIntent)
```

Comment:
>brokerRequest [](start = 67, length = 13)

i think brokerRequest is only used here, and as it's static, can you just create it in this class and use it? no point in pass it around.

Reply:
this is assuming you don't need that correlation_id, which isn't used anywhere

---
In reply to: [570495116](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2380#discussion_r570495116) [](ancestors = 570495116)

Reply:
can you get rid of this method and just put the logic above?

---
In reply to: [570496371](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2380#discussion_r570496371) [](ancestors = 570496371,570495116)

Reply:
It is possible for there to be multiple silent requests happening at once so it is better to keep it as a per request basis. Also, the reason I moved this to its own method is to make the code more readable. Moving low level logic to its own function makes it easier to reuse later and makes reading/understanding the method using it a little bit easier